include/jemalloc/internal/prof_structs.h

```diff
@@ -201,4 +208,14 @@ struct prof_tdata_s {
 };
 typedef rb_tree(prof_tdata_t) prof_tdata_tree_t;
 
+struct prof_recent_s {
+	prof_recent_t *next;
+	size_t usize;
```

Comment:
Is this needed after the allocation is freed? Do we also need to request size in that case?

Reply:
Yes, we always keep the `usize` in the record. I haven't added the request size part yet. That'll be an independent commit / PR later.